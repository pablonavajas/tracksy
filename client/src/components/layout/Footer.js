import React, { Fragment, useContext } from 'react';
import { Link } from 'react-router-dom';

const Footer = () => {
  return (
    // <footer className="footer-class section blue white-text center page-footer">
    //   <div className="container">
    //     <p>Tracksy Copyright &copy; 2020</p>
    //   </div>
    // </footer>

    <footer className="page-footer blue">
      <div className="container">
        <div className="row">
          <div className="col l6 s12">
            <h5 className="white-text">Footer Content</h5>
            <p className="grey-text text-lighten-4">
              You can use rows and columns here to organize your footer content.
            </p>
          </div>
        </div>
      </div>
      <div className="footer-copyright">
        <div className="container">Â© 2020 Tracksy Copyright</div>
      </div>
    </footer>
  );
};

export default Footer;
