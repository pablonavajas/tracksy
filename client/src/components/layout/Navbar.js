import React, { Fragment, useContext } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Navbar = ({ title, icon }) => {
  return (
    <nav className="blue" style={{ marginBottom: '30px' }}>
      <div className="nav-wrapper cont">
        <a href="#" class="brand-logo">
          <i class="material-icons" style={{ marginLeft: '30px' }}>
            {icon}
          </i>
          {title}
        </a>
        <ul id="nav-mobile" className="right hide-on-med-and-down">
          <li className="right">
            <Link to="/register">Register</Link>
          </li>
          <li className="right">
            <Link to="/login">Login</Link>
          </li>
        </ul>
      </div>
    </nav>
    /*
    <Fragment>
      <h1>
        <i className={icon} /> {title}
      </h1>
      <ul>
        <li>
          <Link to="/register">Register</Link>
        </li>
        <li>
          <Link to="/login">Login</Link>
        </li>
      </ul>
    </Fragment>
    */
  );
};

Navbar.propTypes = {
  title: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired
};

Navbar.defaultProps = {
  title: 'Tracksy',
  icon: 'cloud'
};

export default Navbar;
