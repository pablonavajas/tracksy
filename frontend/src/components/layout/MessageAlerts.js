import React, { Fragment, useEffect } from "react";
import { withAlert } from "react-alert";
import { connect } from "react-redux";
import PropTypes from "prop-types";

export const MessageAlerts = ({ message, alert }) => {
  useEffect(() => {
    if (message.succ) {
      alert.success(message.succ);
    } else if (message.fail) alert.error(message.fail);
  });

  return <Fragment />;
};

MessageAlerts.propTypes = {
  message: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
  message: state.messages,
});

export default connect(mapStateToProps)(withAlert()(MessageAlerts));
