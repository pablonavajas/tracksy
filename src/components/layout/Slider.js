import React, { Fragment, useContext } from 'react';
import { Slide } from 'react-slideshow-image';
import { Link } from 'react-router-dom';

const slideImages = ['"../../img/background.jpg'];

const Slider = () => {
  return (
    // <footer className="footer-class section blue white-text center page-footer">
    //   <div className="container">
    //     <p>Tracksy Copyright &copy; 2020</p>
    //   </div>
    // </footer>

    <footer className="page-footer blue">
      <div className="container">
        <div className="row">
          <div className="col l6 s12">
            <h5 className="white-text">Footer Content</h5>
            <p className="grey-text text-lighten-4">
              You can use rows and columns here to organize your footer content.
            </p>
          </div>
        </div>
      </div>
      <div className="footer-copyright">
        <div className="container">Â© 2020 Tracksy Copyright</div>
      </div>
    </footer>
  );
};

export default Footer;
