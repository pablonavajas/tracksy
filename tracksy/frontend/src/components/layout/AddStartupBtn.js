import React from "react";

//Button that triggers AddStartupModal
const AddStartupBtn = () => {
  return (
    <div className="fixed-action-btn">
      <a
        href="#add-startup-modal"
        className="btn-floating  waves-effect waves-light btn-large red darken-1 modal-trigger"
      >
        <i className="large material-icons">add</i>
      </a>
    </div>
  );
};

export default AddStartupBtn;
