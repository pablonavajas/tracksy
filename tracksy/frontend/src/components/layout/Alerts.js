import React, { Component, Fragment } from "react";
import { withAlert } from "react-alert";
import { connect } from "react-redux";
import PropTypes from "prop-types";

export class Alerts extends Component {
  static propTypes = {
    error: PropTypes.object.isRequired,
    message: PropTypes.object.isRequired
  };

  // prevProps (how does this get pulled in)
  componentDidUpdate(prevProps) {
    //if props have changed
    const { error, alert, message } = this.props;
    if (error !== prevProps.error) {
      if (error.msg.name) alert.error(`Name: ${error.msg.name.join()}`);
      else if (error.msg.ownership)
        alert.error(`Ownership: ${error.msg.ownership.join()}`);
      else if (error.msg.website)
        alert.error(`Website: ${error.msg.website.join()}`);
      else if (error.msg.startupEmail)
        alert.error(`Startup Email: ${error.msg.startupEmail.join()}`);
      else if (error.msg.revenue)
        alert.error(`Revenue: ${error.msg.revenue.join()}`);
      else if (error.msg.cashBalance)
        alert.error(`Cash Balance: ${error.msg.cashBalance.join()}`);
      else if (error.msg.monthlyBurn)
        alert.error(`Cash Balance: ${error.msg.monthlyBurn.join()}`);
      else if (error.msg.non_field_errors) {
        alert.error(error.msg.non_field_errors.join());
      } else if (error.msg.username) {
        alert.error(error.msg.username.join());
      }
    }

    if (message !== prevProps.message) {
      if (message.succ) alert.success(message.succ);
      if (message.fail) alert.error(message.fail);
    }
  }

  render() {
    return <Fragment />;
  }
}

const mapStateToProps = state => ({
  error: state.errors,
  message: state.messages
});

export default connect(mapStateToProps)(withAlert()(Alerts));
