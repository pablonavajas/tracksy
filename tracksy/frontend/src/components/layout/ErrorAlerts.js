import React, { Fragment, useEffect } from "react";
import { withAlert } from "react-alert";
import { connect } from "react-redux";
import PropTypes from "prop-types";

function ErrorAlerts({ error, alert }) {
  useEffect(() => {
    if (error.msg.name) alert.error(`Name: ${error.msg.name.join()}`);
    else if (error.msg.ownership)
      alert.error(`Ownership: ${error.msg.ownership.join()}`);
    else if (error.msg.website)
      alert.error(`Website: ${error.msg.website.join()}`);
    else if (error.msg.startupEmail)
      alert.error(`Startup Email: ${error.msg.startupEmail.join()}`);
    else if (error.msg.revenue)
      alert.error(`Revenue: ${error.msg.revenue.join()}`);
    else if (error.msg.cashBalance)
      alert.error(`Cash Balance: ${error.msg.cashBalance.join()}`);
    else if (error.msg.monthlyBurn)
      alert.error(`Cash Balance: ${error.msg.monthlyBurn.join()}`);
    else if (error.msg.non_field_errors) {
      alert.error(error.msg.non_field_errors.join());
    } else if (error.msg.username) {
      alert.error(error.msg.username.join());
    } else if (error.msg && error.msg.length > 0) {
      if (error.msg[0].date) alert.error(error.msg[0].date);
      else if (error.msg[0]) alert.error(error.msg.join());
    } else {
    }
  });

  return <Fragment />;
}

ErrorAlerts.propTypes = {
  error: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  error: state.errors
});

export default connect(mapStateToProps)(withAlert()(ErrorAlerts));
