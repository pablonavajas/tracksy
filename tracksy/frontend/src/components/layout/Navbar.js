import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { logout } from "../../actions/authActions";

const Navbar = ({ title, icon, auth, logout }) => {
  const { isAuthenticated, username, isStartup } = auth;
  const vcLinks = (
    <ul id="nav-mobile" className="right hide-on-med-and-down">
      <span className="mr-3">
        <strong>{username ? `Welcome ${username}` : ""}</strong>
      </span>
      <li className="right">
        <a onClick={logout}>Logout</a>
      </li>
    </ul>
  );

  const startupLinks = (
    <ul id="nav-mobile" className="right hide-on-med-and-down">
      <li>
        <span className="mr-3">
          <strong>{username ? `Welcome ${username}` : ""}</strong>
        </span>
      </li>
      <li className="right">
        <a onClick={logout}>Logout</a>
      </li>
      <li className="right">
        <Link to="/startupPage">History</Link>
      </li>
    </ul>
  );

  const guestLinks = (
    <ul id="nav-mobile" className="right hide-on-med-and-down">
      <li className="right">
        <Link to="/login">Login</Link>
      </li>
      <li className="right">
        <Link to="/register">Register</Link>
      </li>
    </ul>
  );

  return (
    <nav className="blue" style={{ marginBottom: "30px" }}>
      <div className="nav-wrapper cont">
        <a href="/" className="brand-logo">
          <i className="material-icons" style={{ marginLeft: "30px" }}>
            {icon}
          </i>
          {title}
        </a>
        {isAuthenticated ? (isStartup ? startupLinks : vcLinks) : guestLinks}
      </div>
    </nav>
  );
};

Navbar.propTypes = {
  title: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  auth: PropTypes.object.isRequired,
  logout: PropTypes.func.isRequired
};

Navbar.defaultProps = {
  title: "Tracksy",
  icon: "cloud"
};
const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(mapStateToProps, { logout })(Navbar);
