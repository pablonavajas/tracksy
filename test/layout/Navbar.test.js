import React from "react";
import { shallow, mount } from "enzyme";
import { Navbar } from "../../tracksy/frontend/src/components/layout/Navbar";
import ConnectedNavbar from "../../tracksy/frontend/src/components/layout/Navbar";
import toJson from "enzyme-to-json";
import "../setupTests";

//store needed for testing mapStateToProps
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
const mockStore = configureMockStore([thunk]);

describe("Navbar", () => {
  let auth, wrapper, logout;

  beforeEach(() => {
    logout = jest.fn();
    auth = {
      isAuthenticated: true,
      username: "TestingModeUsername",
      isStartup: false,
    };
  });

  it("renders vcLinks, when isAuthenticated is true, and isStartup is false", () => {
    wrapper = shallow(<Navbar auth={auth} logout={logout} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders startupLinks, when isAuthenticated is true, and isStartup is true", () => {
    auth.isStartup = true;
    wrapper = shallow(<Navbar auth={auth} logout={logout} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders guestLinks, when isAuthenticated is false", () => {
    auth.isAuthenticated = false;
    wrapper = shallow(<Navbar auth={auth} logout={logout} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders empty string for username in vcLinks, when username is not provided", () => {
    auth.username = undefined;
    wrapper = shallow(<Navbar auth={auth} logout={logout} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

//Testing MapStateToProps
describe("Navbar ", () => {
  let store, wrapper, isAtuhenticated;

  beforeAll(() => {
    isAtuhenticated = false;
    store = mockStore({ auth: { isAtuhenticated } });
    wrapper = shallow(<ConnectedNavbar store={store} />);
  });

  it("should show the current value imported from store", () => {
    expect(wrapper.props().children.props.auth.isAtuhenticated).toEqual(
      isAtuhenticated
    );
  });
});
