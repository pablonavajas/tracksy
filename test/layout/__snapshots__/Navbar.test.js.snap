// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders empty string for username in vcLinks, when username is not provided 1`] = `
<nav
  className="blue"
  style={
    Object {
      "marginBottom": "30px",
    }
  }
>
  <div
    className="nav-wrapper cont"
  >
    <a
      className="brand-logo"
      href="/"
    >
      <i
        className="material-icons"
        style={
          Object {
            "marginLeft": "30px",
          }
        }
      >
        cloud
      </i>
      Tracksy
    </a>
    <ul
      className="right hide-on-med-and-down"
      id="nav-mobile"
    >
      <span
        className="mr-3"
      >
        <strong />
      </span>
      <li
        className="right"
      >
        <a
          onClick={[MockFunction]}
        >
          Logout
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Navbar renders guestLinks, when isAuthenticated is false 1`] = `
<nav
  className="blue"
  style={
    Object {
      "marginBottom": "30px",
    }
  }
>
  <div
    className="nav-wrapper cont"
  >
    <a
      className="brand-logo"
      href="/"
    >
      <i
        className="material-icons"
        style={
          Object {
            "marginLeft": "30px",
          }
        }
      >
        cloud
      </i>
      Tracksy
    </a>
    <ul
      className="right hide-on-med-and-down"
      id="nav-mobile"
    >
      <li
        className="right"
      >
        <Link
          to="/login"
        >
          Login
        </Link>
      </li>
      <li
        className="right"
      >
        <Link
          to="/register"
        >
          Register
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Navbar renders startupLinks, when isAuthenticated is true, and isStartup is true 1`] = `
<nav
  className="blue"
  style={
    Object {
      "marginBottom": "30px",
    }
  }
>
  <div
    className="nav-wrapper cont"
  >
    <a
      className="brand-logo"
      href="/"
    >
      <i
        className="material-icons"
        style={
          Object {
            "marginLeft": "30px",
          }
        }
      >
        cloud
      </i>
      Tracksy
    </a>
    <ul
      className="right hide-on-med-and-down"
      id="nav-mobile"
    >
      <li>
        <span
          className="mr-3"
        >
          <strong>
            Welcome TestingModeUsername
          </strong>
        </span>
      </li>
      <li
        className="right"
      >
        <a
          onClick={[MockFunction]}
        >
          Logout
        </a>
      </li>
      <li
        className="right"
      >
        <Link
          to="/startupPage"
        >
          History
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Navbar renders vcLinks, when isAuthenticated is true, and isStartup is false 1`] = `
<nav
  className="blue"
  style={
    Object {
      "marginBottom": "30px",
    }
  }
>
  <div
    className="nav-wrapper cont"
  >
    <a
      className="brand-logo"
      href="/"
    >
      <i
        className="material-icons"
        style={
          Object {
            "marginLeft": "30px",
          }
        }
      >
        cloud
      </i>
      Tracksy
    </a>
    <ul
      className="right hide-on-med-and-down"
      id="nav-mobile"
    >
      <span
        className="mr-3"
      >
        <strong>
          Welcome TestingModeUsername
        </strong>
      </span>
      <li
        className="right"
      >
        <a
          onClick={[MockFunction]}
        >
          Logout
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
